### PR TITLE
[ASOC-2886] Add retries when sending graphql requests

### PR DESCRIPTION
**Pull request by samanthamcq**
ASOC-2886 - Fixed an issue with the SDK that was either not working or failing intermittently. Added retries (up to 10 times) for queries, but skips retries for mutations.

### CODE REVIEW

File: finite_state_sdk/__init__.py [hunk 1]
```diff
@@ -1968,6 +1974,7 @@ def search_sbom(token, organization_context, name=None, version=None, asset_vers
     return records
 
 
+@retry(stop=stop_after_attempt(10), wait=wait_fixed(10), retry=retry_if_exception(is_not_breakout_exception))
```
Comment (axelthibodeau) [inline finite_state_sdk/__init__.py:1977]: i think this is too many retries, maybe 3? 5? how long does the test take to run?

File: tests/test_send_graphql_query.py [hunk 2]
```diff
@@ -24,27 +47,88 @@ def test_send_graphql_query_success(self, mock_post):
...
         mock_response = MagicMock()
         mock_response.status_code = 500
         mock_response.text = "Internal Server Error"
         mock_response.json.return_value = {"error": "Internal Server Error"}
         mock_post.return_value = mock_response
 
-        # Call the function and expect an exception
-        with pytest.raises(Exception) as excinfo:
+        # Call the function and expect a RetryError
+        with pytest.raises(tenacity.RetryError) as excinfo:
```
Comment (axelthibodeau) [inline tests/test_send_graphql_query.py:79]: if this test takes too long to run (cause of all the retries) you can add a `send_graphql_query.retry.wait = wait_fixed(0)` in the test itself to override the wait

File: tests/test_send_graphql_query.py [hunk 3]
```diff
@@ -24,27 +47,88 @@ def test_send_graphql_query_success(self, mock_post):
...
             send_graphql_query(self.token, self.organization_context, self.query, self.variables)
 
-        # Assertion
-        assert str(excinfo.value) == "Error: 500 - Internal Server Error"
+        # Check the inner exception message
+        inner_exception = excinfo.value.last_attempt.exception()
```
Comment (axelthibodeau) [inline tests/test_send_graphql_query.py:83]: could also assert the number of calls to `post` here

File: tests/test_send_graphql_query.py (hunk 2 above)
Comment (samanthamcq) [reply tests/test_send_graphql_query.py:79]: It only takes couple minutes.

File: finite_state_sdk/__init__.py (hunk 1 above)
Comment (samanthamcq) [reply finite_state_sdk/__init__.py]: Changed to 5

File: tests/test_send_graphql_query.py (hunk 3 above)
Comment (samanthamcq) [reply tests/test_send_graphql_query.py:83]: added

**Review by axelthibodeau — APPROVED** [review]

**Pull request merged**